tagilmo/utils/malmo_wrapper.py

```diff
@@ -266,6 +270,11 @@ def __init__(self, mc, nAgent = 0):
         if not self.mc.supportsSegmentation():
             self.canBeNone.append('getSegmentationFrame')
         self.cached = {method : (None, 0) for method in self.methods}
+        self.cbuff_history_len = 10
+        self.cached_buffer = {method: (None, 0) for method in self.methods}
+        self.cached_buffer_list = [self.cached_buffer]
```

Comment:
why do you need three caches: self.cached_buffer and self.cached_buffer_list and self.cached? Maybe you can use only cached_buffer_list?

Reply:
Since there were no special methods for processing self.cached from outside and all the work have been done with cache directly, I decided to leave all 3 pieces of data to avoid substantial refactoring. I can rewrite it in unified fashion if it's desirable.